docs/source/reference/api-server/api-server-upgrade.rst

```diff
@@ -0,0 +1,217 @@
+
+Optionally, you can watch the upgrade progress with:
+
+.. code-block:: console
+
+    $ kubectl get pod -l app=${RELEASE_NAME}-api --watch
+    NAME                                       READY   STATUS     RESTARTS   AGE
+    skypilot-demo-api-server-cf4896bdf-62c96   0/1     Init:0/2   0          7s
+    skypilot-demo-api-server-cf4896bdf-62c96   0/1     Init:1/2   0          24s
+    skypilot-demo-api-server-cf4896bdf-62c96   0/1     PodInitializing   0          26s
+    skypilot-demo-api-server-cf4896bdf-62c96   0/1     Running           0          27s
+    skypilot-demo-api-server-cf4896bdf-62c96   1/1     Running           0          50s
+
+The upgraded API server is ready to serve requests after the pod is running and the ``READY`` column shows ``1/1``. The cordon will be removed automatically after the upgrade.
+
+Step 5: Verify the upgrade
+--------------------------
+
+Verify the API server is able to serve requests and the version is consistent with the version you upgraded to:
+
+.. code-block:: console
+
+    $ sky api info
+    Using SkyPilot API server: <URL>
+    ├── Status: healthy, commit: <COMMIT>, version: <VERSION>
+    └── User: <USER> (<USER_ID>)
+
+If possible, you can also trigger your pipelines that depend on the API server to verify there is no compatibility issue after the upgrade.
+
+.. _sky-api-server-vm-upgrade:
+
+Upgrade the API server deployed on VM
+-------------------------------------
+
+.. note::
+
+    VM deployment does not offer graceful upgrading support. We recommend to use Helm deployment :ref:`sky-api-server-deploy` in production environments.
+
+Suppose the cluster name of the API server is ``api-server`` (which is used in the :ref:`sky-api-server-cloud-deploy` guide), you can upgrade the API server with the following steps:
+
+1. Switch to the original API server that is used to launch the API server VM instance. It is usually the local API server that was launched automatically when you ran ``sky launch`` to launch the API server VM instance in the :ref:`sky-api-server-cloud-deploy` guide:
+
+.. code-block:: bash
+
+    # Replace http://localhost:46580 with the real API server endpoint if you were not using the local API server to launch the API server VM instance.
+    sky api login -e http://localhost:46580
+
+2. Check the API server VM instance is ``UP``:
+
+.. code-block:: console
+
+    $ sky status --no-show-managed-jobs --no-show-services
+    Clusters
+    NAME        LAUNCHED     RESOURCES                                                                  STATUS  AUTOSTOP  COMMAND
+    api-server  41 mins ago  1x AWS(c6i.2xlarge, image_id={'us-east-1': 'docker:berkeleyskypilot/sk...  UP      -         sky exec api-server pip i...
+
+3. Upgrade the SkyPilot pip package on the VM and restart the API server:
+
+.. note::
+
+    Upgrade and restart the API server will interrupt all running requests and may cause compatibility issue that requires upgrading the clients to resolve.
```

Comment:
```suggestion
    Upgrade and restart the API server will interrupt all running requests.
```

Reply:
Instead of talking about backward compatibility here, how about we just add another step to upgrade all clients, just like the helm deployment upgrade guide.

Reply:
Actually, there is little gain to explicitly upgrade all clients without draining. Because:

- If there is no break change, upgrading is not actually needed;
- If there is a break change, upgrading client break the communication between client/server immediately;

But sure we can add the step and warn users about this

Reply:
Ahh, when we talk about upgrading the clients, we should mention that if cordon and drain is not done, clients should wait until the API server to finish upgrade manually. We can have some sentences like the following in the client upgrade section, so that we can keep the cordon & drain section to be at the end. Reason for keeping it later: we want to avoid the feeling of complexity for non-advanced users, but also enable the advanced user use case.

```
After upgrading the clients, they should avoid being used until the API server is upgraded to the new version. To enforce the cordon and drain of the old API server, you can refer to :ref:`section-to-cordon-and-drain`.
```

Wdyt?

Reply:
Sounds good!